src/ops/binary_ops.rs

```diff
@@ -30,7 +30,9 @@ impl Op for BinaryOps {
     }
 
     fn propagate(&self, source: &Value) {
-        let grad = &source.grad().unwrap();
+        let grad = &source
+            .grad()
+            .unwrap_or_else(|| panic!("Cannot backpropagate when gradient is None"));
```

Comment:
In the future we would have custom error handling but for now panicking should suffice